src/energyplus/ForwardTranslator/ForwardTranslateZoneHVACTerminalUnitVariableRefrigerantFlow.cpp

```diff
@@ -264,121 +249,138 @@ namespace energyplus {
       }
     }
 
-    boost::optional<HVACComponent> _mo_supplementalHeatingCoil = modelObject.supplementalHeatingCoil();
-
-    if (boost::optional<IdfObject> _fan = translateAndMapModelObject(fan)) {
-      // SupplyAirFanObjectType
-
-      idfObject.setString(ZoneHVAC_TerminalUnit_VariableRefrigerantFlowFields::SupplyAirFanObjectType, _fan->iddObject().name());
-
-      // SupplyAirFanObjectName
-
-      idfObject.setString(ZoneHVAC_TerminalUnit_VariableRefrigerantFlowFields::SupplyAirFanObjectName, _fan->name().get());
+    if (auto heatingCoil = modelObject.heatingCoil()) {
+      if (boost::optional<IdfObject> _heatingCoil = translateAndMapModelObject(heatingCoil.get())) {
 
-      std::string fanInletNodeName;
+        idfObject.setString(ZoneHVAC_TerminalUnit_VariableRefrigerantFlowFields::HeatingCoilObjectType, _heatingCoil->iddObject().name());
 
-      if (heatingCoil) {
-        fanInletNodeName = heatOutletNodeName;
-      } else if (coolingCoil) {
-        fanInletNodeName = coolOutletNodeName;
-      } else if (translateMixer()) {
-        fanInletNodeName = mixerOutletNodeName;
-      } else {
-        fanInletNodeName = inletNodeName;
-      }
+        idfObject.setString(ZoneHVAC_TerminalUnit_VariableRefrigerantFlowFields::HeatingCoilObjectName, _heatingCoil->name().get());
 
-      // If no supplemental coil, then directly the outletNodeName, otherwise it has its own name constructed above
-      if (!_mo_supplementalHeatingCoil) {
-        fanOutletNodeName = outletNodeName;
+        compsInOrder.emplace_back(Component(_heatingCoil.get(), modelObject.name().get() + " Heating Coil Outlet Node",
+                                            Coil_Heating_DX_VariableRefrigerantFlowFields::CoilAirInletNode,
+                                            Coil_Heating_DX_VariableRefrigerantFlowFields::CoilAirOutletNode));
       }
+    }
 
-      if (fan.iddObject().type() == IddObjectType::OS_Fan_OnOff) {
-        _fan->setString(Fan_OnOffFields::AirInletNodeName, fanInletNodeName);
-        _fan->setString(Fan_OnOffFields::AirOutletNodeName, fanOutletNodeName);
-      } else if (fan.iddObject().type() == IddObjectType::OS_Fan_ConstantVolume) {
-        _fan->setString(Fan_ConstantVolumeFields::AirInletNodeName, fanInletNodeName);
-        _fan->setString(Fan_ConstantVolumeFields::AirOutletNodeName, fanOutletNodeName);
-      } else if (fan.iddObject().type() == IddObjectType::OS_Fan_SystemModel) {
-        _fan->setString(Fan_SystemModelFields::AirInletNodeName, fanInletNodeName);
-        _fan->setString(Fan_SystemModelFields::AirOutletNodeName, fanOutletNodeName);
-      } else {
-        LOG(Error, "VRF named " << modelObject.name().get() << " uses an unsupported fan type.");
+    if (!t_blowThrough) {
+      // For now this is always instantiated...
+      if (boost::optional<HVACComponent> fan_ = modelObject.supplyAirFan()) {
+        translateFan(fan_.get());
```

Comment:
Option 2: when drawthrough, it's after Cooling/Heating Coils (if any), before Supplemental Heating Coil (if any)